### PR TITLE
Changing the payment status of a sales invoice

### PR DESCRIPTION
**Pull request by arek1-618**
Changing the payment status of a sales invoice

### CODE REVIEW

File: modules/PaymentsIn/helpers/FinvoicePaymentStatus.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * @copyright YetiForce Sp. z o.o.
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+
+/**
+ * Class to change the payment status of a sales invoice.
+ */
+class PaymentsIn_FinvoicePaymentStatus_Helper
```
Comment (rskrzypczak) [inline modules/PaymentsIn/helpers/FinvoicePaymentStatus.php]: remove halper file

File: modules/PaymentsIn/models/Module.php [hunk 2]
```diff
@@ -14,61 +14,30 @@
...
 	 * Get the sum of all payments by record ID.
 	 *
-	 * @param int $recordId
+	 * @param int    $recordId
+	 * @param string $moduleName
 	 *
 	 * @return float
 	 */
-	private static function getSumOfPaymentsByRecordId(int $recordId): float
+	public static function getSumOfPaymentsByRecordId(int $recordId, string $moduleName): float
```
Comment (mariuszkrzaczkowski) [inline modules/PaymentsIn/models/Module.php]: move to PaymentsIn_PaymentStatus_Model

File: modules/PaymentsIn/models/PaymentStatus.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	/**
+	 * Checking if you can update the payment status.
+	 *
+	 * @param Vtiger_Record_Model $recordModel
+	 *
+	 * @return bool
+	 */
+	protected static function canUpdatePaymentStatus(Vtiger_Record_Model $recordModel): bool
+	{
+		$fieldModel = \Vtiger_Module_Model::getInstance(static::$moduleName)->getFieldByName(static::$fieldPaymentStatusName);
```
Comment (mariuszkrzaczkowski) [inline modules/PaymentsIn/models/PaymentStatus.php]: optimization: 
check first
```
!$recordModel->isEmpty(static::$relatedRecordIdName)
($recordModel->isNew() || false !== $recordModel->getPreviousValue(static::$fieldPaymentStatusName))
```

**Comment by piotr-siedlak** [comment]
Hi,
how I can implement this change in easy way?  In 5.2 I can't change payment status.

**Pull request merged**